### ISSUE TITLE
multigpu training does not appear to be running

### ISSUE THREAD

**Issue by nikky4D** [MaximIntegratedAI/ai8x-training#231]
how do I start multigpu training? I have tried to use --gpus 0,1,2 or set the CUDA_VISIBLE_DEVICES=0,1,2. However, the model appears to only be running on the first specified gpu (in this case 0). Am I using this correctly?

**Comment by rotx-maxim** [comment]
Could you add the complete command line please? You shouldn't have to specify anything, by default all available resources are used.
What does nvidia-smi show during training?
In some cases, the data loader may be the limiting factor, in particular if the dataset needs to be "prepared" first. In that case, wait until "Epoch 0" shows up.
Another thing that may help finding issues like this is to train the MNIST example.

**Comment by nikky4D** [comment]
I am using this command for the mscoco dataset, using the newly added fpndetector ```  #!/bin/sh
python train.py --deterministic --print-freq 50 --out-dir mscoco-train --epochs 400 --gpus 1,2,3 --workers 4 --optimizer Adam --lr 0.001 --wd 0 --model ai87fpndetector --use-bias --momentum 0.9 --weight-decay 1e-6 --dataset mscoco_256_320_augmented --device MAX78002 --obj-detection --obj-detection-params parameters/obj_detection_params_pascalvoc.yaml --batch-size 32 --qat-policy policies/qat_policy_pascalvoc.yaml --compress policies/schedule-pascalvoc.yaml --validation-split 0 "$@" ``` 

Currently, nvidia-smi shows only GPU1 being used

**Comment by nikky4D** [comment]
I've trained with train_mnist,sh and train_cifar.sh and train_pascalvoc_fpndetector.sh. MNIST and CIFAR both train with multigpu. But the pascalvoc does not use multigpu. I've added --gpus flag to mnist and I can control the gpus that are used for multigpu training. But in pascalvoc train, with or without the gpus flag, I'm still using one gpu.

**Comment by rotx-maxim** [comment]
Thanks for your report! We're looking into it.

**Comment by nikky4D** [comment]
Hi, is there any update on this? Am I making an error in how I call the train function?

**Comment by rotx-maxim** [comment]
Could you try #233?
Please note that requirements*.txt has changed, and that while training uses all GPUs, mAP calculation during verification is running on a single GPU.

**Comment by rotx-maxim** [comment]
Could you post the exception printout please?

**Comment by nikky4D** [comment]
Here it is: 
train_multi.py is a copy of the train.py file in the PR
```
Training epoch: 9431 samples (128 per mini-batch)
Epoch: [0][   50/   74]    Overall Loss 12.860596    Objective Loss 12.860596                                        LR 0.000100    Time 1.245014    
Epoch: [0][   74/   74]    Overall Loss 12.006912    Objective Loss 12.006912                                        LR 0.000100    Time 1.111727    
--- validate (epoch=0)-----------
1047 samples (128 per mini-batch)
{'multi_box_loss': {'alpha': 2, 'neg_pos_ratio': 3}, 'nms': {'min_score': 0.2, 'max_overlap': 0.45, 'top_k': 5}}
Traceback (most recent call last):
  File "train_multi.py", line 1755, in <module>
    main()
  File "train_multi.py", line 549, in main
    top1, top5, vloss, mAP = validate(val_loader, model, criterion, [pylogger],
  File "train_multi.py", line 906, in validate
    return _validate(val_loader, model, criterion, loggers, args, epoch, tflogger)
  File "train_multi.py", line 1091, in _validate
    filtered_all_images_boxes, \
ValueError: not enough values to unpack (expected 3, got 0)


Log file for this run: /home/ai8x-training/flir-train-June9/lr_1e4_wd_5e6_ep2300___2023.06.13-160457/lr_1e4_wd_5e6_ep2300___2023.06.13-160457.log
Traceback (most recent call last):
  File "train_multi.py", line 1755, in <module>
    main()
  File "train_multi.py", line 549, in main
    top1, top5, vloss, mAP = validate(val_loader, model, criterion, [pylogger],
  File "train_multi.py", line 906, in validate
    return _validate(val_loader, model, criterion, loggers, args, epoch, tflogger)
  File "train_multi.py", line 1091, in _validate
    filtered_all_images_boxes, \
ValueError: not enough values to unpack (expected 3, got 0)
```

**Comment by rotx-maxim** [comment]
I've updated the code. Please let me know whether this fixes the issue!

**Comment by nikky4D** [comment]
Thank you. Multi-gpu appears to be running now with no errors. I've gone through 10 epochs and see no errors

**Comment by rotx-maxim** [comment]
Fixed in #233
